### PR TITLE
[stable/grafana] Add sidecar options for env and useCurrentNamespace

### PR DESCRIPTION
**Pull request by eloo**
Adding values for configuring the sidecar with additional envs or use the current namespace of the pods for configmap lookups.

Related to: https://github.com/kiwigrid/k8s-sidecar/issues/2

### CODE REVIEW

**Comment by eloo** [comment]
CLA signed

**Comment by stealthybox** [comment]
It would be good to downgrade the ClusterRole and ClusterRoleBinding to a Role and RoleBinding when using this feature, as the sidecar shouldn't need to be able to read ConfigMaps across the entire cluster.

**Comment by stealthybox** [comment]
The sidecar would not be able to get ConfigMaps from the Kubernetes API without a permitting RoleBinding or ClusterRoleBinding.

If it's working, you may have some leftover object from a previous installation.
It's also common misadvice to bind very permissive roles to the default ServiceAccount.
If you've modified the default SA in the namespace you're testing in, this may be where your API permissions are coming from.

**Comment by eloo** [comment]
@stealthybox 
thanks for clarification but i'm not that familiar with the rbac concept.
i hope my changes are going into the correct way to fit your needs ;)

File: stable/grafana/templates/clusterrole.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {{- if .Values.rbac.create }}
+{{- if and (or .Values.sidecar.dashboards.enabled .Values.sidecar.datasources.enabled) .Values.sidecar.useCurrentNamespace  }}
```
Comment (stealthybox) [inline stable/grafana/templates/clusterrole.yaml]: this line can probably just be:
```
{{- if .Values.sidecar.useCurrentNamespace  }}
```

The entire file should probably be wrapped in:
```
{{- if and .Values.rbac.create (or .Values.sidecar.dashboards.enabled .Values.sidecar.datasources.enabled) }}
```

File: stable/grafana/templates/clusterrolebinding.yaml [hunk 2]
```diff
@@ -1,8 +1,16 @@
 {{- if .Values.rbac.create }}
+{{- if and (or .Values.sidecar.dashboards.enabled .Values.sidecar.datasources.enabled) .Values.sidecar.useCurrentNamespace  }}
```
Comment (stealthybox) [inline stable/grafana/templates/clusterrolebinding.yaml]: same as above

**Review by stealthybox — CHANGES_REQUESTED** [review]
Thanks @eloo -- this is totally what I was thinking

Here's something I noticed on the conditionals:

**Comment by eloo** [comment]
@stealthybox okay i've adapted my PR

**Comment by stealthybox** [comment]
/lgtm

@mattfarina would you mark this as `/ok-to-test` ?
I'm not in the Helm org since we moved things over.

**Comment by stealthybox** [comment]
/assign @linki 

got a sec for this one @linki ? :)

**Comment by giacomoguiulfo** [comment]
@eloo thanks for the effort!

The k8s-sidecar now supports lookup in the current namespace "natively", and `AFAIK`, there is no need to make the chart more complex by adding `RBAC` and `ClusterRole` resources because the pod no longer needs them to get the namespace information.

Take a look at #7702. IMHO, I think this PR should perhaps drop that part of the changes. However, the additonal envs feature can still go as part of this PR 👍

**Review by giacope — CHANGES_REQUESTED** [review]
👆

**Comment by eloo** [comment]
@giacomoguiulfo okay so i will close this PR

**Comment by giacomoguiulfo** [comment]
@stealthybox You are correct, my bad 😞. We should have this changes in, (ping @eloo). Perhaps, the only modification will be the use of `useCurrentNamespace`, depending on what do you guys think is the most appropiate way for configuring that (either letting the sidecar figure it out by not passing an env variable or by setting `NAMESPACE` to `fieldPath: metadata.namespace`).

However, for the case of using the current namespace, the RBAC resources can still be optional since the default `ServiceAccount` is enough for the task. This doesn't affect the changes though.

**Comment by stealthybox** [comment]
We should explicitly define the Role.

We shouldn't rely on a namespace's configured default SA, as it's managed outside the chart.
(It may be modified by a user/operator or another chart)

The Role for the current namespace use-case would also be more specific than most default SA's.
( can only read configmaps )

**Comment by eloo** [comment]
@stealthybox so whats the plan?

**Comment by stealthybox** [comment]
@eloo I think your patch is good and can be merged right now. (cc @davidkarlsen)

#7702 needs some edits and could benefit from your work.

**Comment by stealthybox** [comment]
@eloo see instructions for signing off on your commits: https://github.com/helm/charts/pull/7060/checks?check_run_id=18506998

**Comment by eloo** [comment]
something went wrong with the signoff step.. i will reopen this PR

**Pull request closed without merging**